ssl/ssl_quic.c

```diff
@@ -114,15 +114,26 @@ int SSL_provide_quic_data(SSL *ssl, OSSL_ENCRYPTION_LEVEL level,
         QUIC_DATA *qd;
         const uint8_t *p = data + 1;
 
+        /* Check for an incomplete block */
+        qd = ssl->quic_input_data_tail;
+        if (qd != NULL) {
+            l = ssl->quic_input_data_tail->length - ssl->quic_input_data_tail->offset;
```

Comment:
Drive-by comment: why not `l = qd->length - qd->offset`?

Reply:
Because I did this really "quic" this morning. Ha! Yes, that would be better. Still a WIP. I introduced the `qd` variable late in the editing of this.

Reply:
I think you need to check that whether the message is incomplete or not in quic_get_message.  If it is incomplete, it should not be processed otherwise we get some sort of invalid message error.